src/mds/MDSRank.cc

```diff
@@ -330,6 +330,11 @@ class C_Drop_Cache : public MDSInternalContext {
         recall_client_state();
       });
       mds->timer.add_event_after(1.0, timer);
+      if (gather->has_subs()) {
+        gather->activate();
```

Comment:
This will assert since finisher is still not set. There needs to be a `->set_finisher()` followed by `->activate()`.